Bottom: 8aa0b46a53b12878f7bb4b3e1bcf9eee6f38e208
Top:    e87c262ad68d196199970049bc948f612ff5b721
Author: Andrew Cooper <andrew.cooper@novatechweb.com>
Date:   2018-07-12 11:12:22 -0500

Add buildbot role and new playbook


---
diff --git a/ansible-playbook/buildbot.yml b/ansible-playbook/buildbot.yml
new file mode 100644
index 00000000..68181dcc
--- /dev/null
+++ b/ansible-playbook/buildbot.yml
@@ -0,0 +1,14 @@
+---
+
+- name: Install/Setup Docker images and containers on Daedalus
+  hosts: daedalus
+  become: true
+  become_user: root
+  become_method: sudo
+  tags:
+  - buildbot
+  roles:
+  - name: Build and start the Buildbot master container
+    role: buildbot-master
+    tags:
+    - buildbot-master
diff --git a/ansible-playbook/group_vars/all/buildbot.yml b/ansible-playbook/group_vars/all/buildbot.yml
new file mode 100644
index 00000000..86476751
--- /dev/null
+++ b/ansible-playbook/group_vars/all/buildbot.yml
@@ -0,0 +1,3 @@
+---
+buildbot_version: 1.1.2
+buildbot_worker_port: 9989
diff --git a/ansible-playbook/group_vars/all/credentials.yml b/ansible-playbook/group_vars/all/credentials.yml
index c306cadf..d476bf91 100644
--- a/ansible-playbook/group_vars/all/credentials.yml
+++ b/ansible-playbook/group_vars/all/credentials.yml
@@ -1,5 +1,6 @@
 ---
 # passwords
+buildbot_admin_password:             '{{ lookup("password", playbook_dir + "/credentials/buildbot/admin"                      , length=20) }}'
 buildsystem_database_build_password: '{{ lookup("password", playbook_dir + "/credentials/buildsystem/database/readuser") }}'
 buildsystem_database_root_password:  '{{ lookup("password", playbook_dir + "/credentials/buildsystem/database/root"           , length=20) }}'
 buildsystem_ssh_prod_user_password:  '{{ lookup("password", playbook_dir + "/credentials/buildsystem/ssh/prod"                , length=20) }}'
diff --git a/ansible-playbook/group_vars/docker-hosts/networking.yml b/ansible-playbook/group_vars/docker-hosts/networking.yml
index f908c8f0..42db8fdb 100644
--- a/ansible-playbook/group_vars/docker-hosts/networking.yml
+++ b/ansible-playbook/group_vars/docker-hosts/networking.yml
@@ -47,3 +47,9 @@ phpldapadmin_port_args:
 - "{{ phpldapadmin_ip_addr }}:443:443"
 
 phpmyadmin_hostname: phpmyadmin.novatech-llc.com
+
+buildbot_hostname: buildbot.novatech-llc.com
+buildbot_port_args:
+- "{{ buildbot_ip_addr }}:80:8080"
+- "{{ buildbot_ip_addr }}:443:8443"
+- "{{ buildbot_ip_addr }}:9989:9989"
diff --git a/ansible-playbook/host_vars/daedalus.novatech-llc.com/networking.yml b/ansible-playbook/host_vars/daedalus.novatech-llc.com/networking.yml
index c6323198..5e2d2774 100644
--- a/ansible-playbook/host_vars/daedalus.novatech-llc.com/networking.yml
+++ b/ansible-playbook/host_vars/daedalus.novatech-llc.com/networking.yml
@@ -15,3 +15,4 @@ exim4_ip_addr:        172.16.0.107
 openldap_ip_addr:     172.16.0.108
 phpldapadmin_ip_addr: 172.16.0.108
 phpmyadmin_ip_addr:   172.16.0.109
+buildbot_ip_addr:     172.16.0.110
diff --git a/ansible-playbook/host_vars/testdaedalus.novatech-llc.com/networking.yml b/ansible-playbook/host_vars/testdaedalus.novatech-llc.com/networking.yml
index b3e366ce..aa95e93b 100644
--- a/ansible-playbook/host_vars/testdaedalus.novatech-llc.com/networking.yml
+++ b/ansible-playbook/host_vars/testdaedalus.novatech-llc.com/networking.yml
@@ -15,3 +15,4 @@ exim4_ip_addr:        172.16.103.107
 openldap_ip_addr:     172.16.103.108
 phpldapadmin_ip_addr: 172.16.103.108
 phpmyadmin_ip_addr:   172.16.103.109
+buildbot_ip_addr:     172.16.103.110
diff --git a/ansible-playbook/site.yml b/ansible-playbook/site.yml
index 0727917d..ee706b65 100644
--- a/ansible-playbook/site.yml
+++ b/ansible-playbook/site.yml
@@ -90,6 +90,9 @@
     tags:
     - buildsystem_container
 
+- name: Setup the buildbot containers
+  import_playbook: buildbot.yml
+
 - name: Setup the Build System (Test Station) with repo data
   hosts: buildsystem
   remote_user: ansibleremote
diff --git a/docker/buildbot-master/Dockerfile b/docker/buildbot-master/Dockerfile
index 521c8237..2b04303b 100644
--- a/docker/buildbot-master/Dockerfile
+++ b/docker/buildbot-master/Dockerfile
@@ -1,4 +1,4 @@
-FROM        buildbot/buildbot-master:v0.9.13
+FROM        buildbot/buildbot-master:v1.2.0
 MAINTAINER  Andrew Cooper <andrew.cooper@novatechweb.com>
 
 WORKDIR /buildbot
diff --git a/docker/buildbot-master/buildbot/master.cfg b/docker/buildbot-master/buildbot/master.cfg
index 658d5fb1..75985b8a 100644
--- a/docker/buildbot-master/buildbot/master.cfg
+++ b/docker/buildbot-master/buildbot/master.cfg
@@ -68,8 +68,8 @@ c['www'] = {
 # installation's html.WebStatus home page (linked to the
 # 'titleURL') and is embedded in the title of the waterfall HTML page.
 
-c['title'] = "workspace"
-c['titleURL'] = "http://wiki"
+c['title'] = "NovaTech CI"
+c['titleURL'] = "http://git.novatech-llc.com/"
 
 # the 'buildbotURL' string should point to the location where the buildbot's
 # internal web server (usually the html.WebStatus page) is visible. This
diff --git a/roles/buildbot-master/defaults/main.yml b/roles/buildbot-master/defaults/main.yml
new file mode 100644
index 00000000..c0988cfb
--- /dev/null
+++ b/roles/buildbot-master/defaults/main.yml
@@ -0,0 +1,93 @@
+---
+# defaults file for buildbot
+
+# host containing build assets
+asset_host: 'http://{{ansible_facts.default_ipv4.address}}/'
+
+# backup host directory
+backup_hostdir: '{{ docker_backup_dir }}/buildbot'
+
+# numeric ID of buildbot user
+buildbot_uid: 1000
+
+# numeric version of buildbot service
+buildbot_version: '1.1.2'
+
+# port number for workers to communicate with master
+buildbot_worker_port: 9989
+
+# port number for unencrypted web access
+buildbot_http_port: 8080
+
+# port number for encrypted web access
+buildbot_https_port: 8443
+
+# The hostname passed as an envirnment variable into the container
+buildbot_ip_addr: '127.0.0.1'
+buildbot_hostname: buildbot.example.com
+buildbot_port_args:
+  - '{{buildbot_http_port}}'
+  - '{{buildbot_https_port}}'
+  - '{{buildbot_worker_port}}'
+
+# host path containing configuration files
+config_hostdir: '{{ docker_restore_config_base_dir }}/{{ container_name }}'
+
+# environment passed to the container
+container_env:
+  BUILDBOT_DATABASE: 'sqlite:///{{ data_path }}/state.sqlite'
+  BUILDBOT_WEB_PORT: '{{buildbot_http_port}}'
+  BUILDBOT_WEB_URL: 'http://{{ buildbot_hostname }}/'
+  BUILDBOT_WORKER_PORT: '{{buildbot_worker_port}}'
+  BUILDBOT_WORKER_NTEL: 'worker-ntel'
+  BUILDBOT_WORKER_NTEL_PASS: '{{ buildbot_worker_ntel_password }}'
+  BUILDBOT_WORKER_PTXDIST: 'worker-ptxdist'
+  BUILDBOT_WORKER_PTXDIST_PASS: '{{ buildbot_worker_ptxdist_password }}'
+
+# the name of the container being started
+container_name: '{{ docker_name_prefix }}buildbot'
+
+# networks of the container
+container_networks:
+  - name: '{{ docker_network_frontend }}'
+
+# volumes mounted within the container
+container_volumes:
+  - '{{ data_volume }}:{{ data_path }}:z'
+  - '{{ config_hostdir }}:{{ secrets_path }}'
+
+# mount path of the data volume
+data_path: '/buildbot'
+
+# name of the data volume
+data_volume: '{{ docker_name_prefix }}buildbot_DV'
+
+# hostname
+hostname: buildbot-worker-ntel
+
+# build arguments of the image
+image_args:
+  BUILDBOT_UID: '{{buildbot_uid}}'
+  BUILDBOT_VERSION: 'v{{buildbot_version}}'
+  BUILDBOT_TIMEZONE: '{{ansible_facts.date_time.tz}}'
+
+# directory of the image source
+image_dir: '{{ docker_projects_dir }}/buildbot-master'
+
+# name of the image being built
+image_name: '{{ docker_registry_username }}/buildbot'
+
+# repository URI of the image source
+image_repo: https://github.com/novatechweb/docker-buildbot-master.git
+
+# restoration host directory
+restore_hostdir: '{{ bacula_dest }}{{ backup_hostdir }}'
+
+# host path containing secrets
+secrets_hostdir: '{{ config_hostdir }}'
+
+# name of the ssh known_hosts secret
+secrets_known_hosts: 'known_hosts'
+
+# mount path of the secrets volume
+secrets_path: '/run/secrets'
diff --git a/roles/buildbot-master/tasks/main.yml b/roles/buildbot-master/tasks/main.yml
new file mode 100644
index 00000000..63fac665
--- /dev/null
+++ b/roles/buildbot-master/tasks/main.yml
@@ -0,0 +1,41 @@
+---
+# tasks file for buildbot
+
+# This will generate passwords for these accounts.
+- assert:
+    that:
+    - buildbot_admin_password is defined
+
+- name: Create buildbot data volume
+  docker_volume:
+    name: '{{ data_volume }}'
+
+- name: Checkout image repo
+  git:
+    repo: '{{ image_repo }}'
+    version: master
+    dest: '{{ image_dir }}'
+
+- name: Create buildbot master image
+  docker_image:
+    name: '{{ image_name }}'
+    tag: '{{ docker_image_tag }}'
+    path: '{{ image_dir }}'
+    buildargs: '{{ image_args }}'
+    force: true
+
+- name: Start buildbot container
+  docker_container:
+    name: '{{ container_name }}'
+    image: '{{ image_name }}:{{ docker_image_tag }}'
+    networks: '{{ container_networks }}'
+    volumes: '{{ container_volumes }}'
+    env: '{{ container_env }}'
+    ports: '{{ buildbot_port_args }}'
+    state: started
+
+- name: Add buildbot_master host
+  add_host:
+    name: buildbot_master
+    pubkey: "{{ output.stdout }}"
+    ip_addr: "{{ buildbot_ip_addr }}"
diff --git a/roles/buildbot-master/vars/main.yml b/roles/buildbot-master/vars/main.yml
new file mode 100644
index 00000000..eed3963e
--- /dev/null
+++ b/roles/buildbot-master/vars/main.yml
@@ -0,0 +1,2 @@
+---
+# vars file for buildbot
